kernel/init.c

```diff
@@ -219,18 +227,18 @@ static void bg_thread_main(void *unused1, void *unused2, void *unused3)
 	z_stack_adjust_initialized = 1;
 #endif
 	if (boot_delay > 0 && IS_ENABLED(CONFIG_MULTITHREADING)) {
-		printk("***** delaying boot " STRINGIFY(CONFIG_BOOT_DELAY)
-		       "ms (per build configuration) *****\n");
+		immediate_printk("*** Delaying boot %u ms ***\n",
+			boot_delay);
 		k_busy_wait(CONFIG_BOOT_DELAY * USEC_PER_MSEC);
```

Comment:
shouldn't you use `boot_delay` here ? otherwise it will use whatever was set in CONFIG_BOOT_DELAY and not the randomized value.

Reply:
True. Corrected. As well the time units have been wrong. Corrected now as well.